### ISSUE TITLE
Notebook list not updating

### ISSUE THREAD

**Issue by jjon** [laurent22/joplin#5381]
dear all,
I'm running Joplin 2.3.5 on both my mac (10.12.6) and my linux laptop (Ubuntu 5.11 kernel, popOS). I updated both clients recently. I don't know when or how it happened, but I finally noticed that while all my notes are synced (via WebDAV), my notebook list is not. Several of the notebooks that show on my Linux client are missing on my Mac client's notebook list. I can find all the notes on either client via search or tag, but the Linux client shows 14 notebooks while the Mac shows only 9. Using find, I see the note and it's correctly identified as belonging to the right notebook, but on the mac, that notebook nevertheless does not appear in the list.

Since I'm not sure when it happened, I can't reproduce it. My surmise is that during the period when the mac client was running a more up-to-date version than the linux client, new notebooks created on the linux did not appear in the mac notebook list. Now that the linux client is updated, new notebooks created there do appear on the mac notebook list (and vice versa) when I sync.

In the mac console, two errors appear immediately on startup. They don't appear to be related, but whatdoiknow.

```console.log
/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/Logger.js:188
11:16:09: RevisionService: RevisionService::collectRevisions: Could not create revision for note: 4a3369b1f14b4b188268532617232d8f: SyntaxError: Unexpected token in JSON at position 105
    at JSON.parse (<anonymous>)
    at Function.applyTextPatch (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/models/Revision.js:65:49)
    at Function.<anonymous> (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/models/Revision.js:269:36)
    at Generator.next (<anonymous>)
    at fulfilled (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/models/Revision.js:5:58)
```
and
```console.log
/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/Logger.js:188
11:16:09: RevisionService: RevisionService::maintenance: Error: Invalid patch string: @@ -568,9 +568,8 @@\n 4991cc45\n-)\n
    at diff_match_patch.patch_fromText (node_modules/@joplin/lib/node_modules/diff-match-patch/index.js:2092)
    at Function.applyTextPatchLegacy (node_modules/@joplin/lib/models/Revision.js:43)
    at Function.applyTextPatch (node_modules/@joplin/lib/models/Revision.js:59)
    at Function.<anonymous> (node_modules/@joplin/lib/models/Revision.js:269)
    at Generator.next (<anonymous>)
    at fulfilled (node_modules/@joplin/lib/models/Revision.js:5)
```
These two errors always appear upon sync.

Is there some kind of incantation I can invoke to force the mac to update its notebook list?

befuddled,
Jon

**Comment by jjon** [comment]
OK. No suggestions from the peanut gallery?
A little further naive tinkering in the DevTools console and I learn:
`Folder.all()` returns an array of 15 items: all my Notebooks, but `Folder.allAsTree()` returns only 11. It would appear that I have 4 folders that are not part of the "tree".

So, I tried rebooting the whole system and Joplin launched into a massive update that eventually took an hour and a quarter. When it had finally finished, The Mac desktop app Notebook List still only lists 11 out of 15 items.

Moreover: when I start up the Android and the Linux clients, they went and fetched **everything**, and now `Folder.all()` on the Linux client returns 16 items, `Folder.allAsTree()` returns 12 items, and the Notebook List shows an additional "Welcome! (Desktop)" notebook. *huh?*
 
So, still trying to get the Notebook List to show all the existing folders. Something I'm not getting about how this is supposed to work? Are all folders Notebooks?

still confused,
Jon

**Comment by Daeraxa** [comment]
I can't answer many of your questions here but in terms of getting the info sync'd up it might be worth trashing the mac database entirely, importing a fresh JEX export file from your linux system and setting up your sync on mac again.
Alternatively there is an option in the sync settings page to force a full replacement download from the sync target that will replace all the data on your client.

**Comment by jjon** [comment]
Hi there stalebot, no, the issue has not been fixed. It still does require the community's attention. I've not yet turned to the drastic measures suggested by @Daeraxa, I'd much rather first understand the nature of the relationship between Folder and Notebook, and how and why mine grew apart. That will take some time. Meanwhile, the ordinary functioning of Joplin seems unperturbed, I just go along knowing that there are Folders that can never be Notebooks: so sad.

**Comment by jjon** [comment]
update. thanks to @roman-r-m I was able to fix my immediate problem. The folders not appearing in the notebook list all had a `parent_id` of a nonexistent notebook in the sqlite db. I simply cleared the `parent_id` field for all those folders, and restarted Joplin. The offending folders then appeared in the Notebooks list. I renamed them in the Joplin UI, synced, and the newly renamed folders appeared in my other clients as well.

If there was a bug that created this problem, it was some time ago, and I can't now reproduce it, so I'm closing this issue.
